ivy/func_wrapper.py

```diff
@@ -45,6 +47,34 @@ def _get_first_array(*args, **kwargs):
 # ---------------#
 
 
+def handle_array_like(fn: Callable) -> Callable:
+
+    @functools.wraps(fn)
+    def new_fn(*args, **kwargs):
+        args = list(args)
+        parameters = list(typing.get_type_hints(fn))
```

Comment:
calling `typing.get_type_hints(fn)` twice is very inefficient. You should cache this first: `type_hints = typing.get_type_hints(fn)`